src/Container/AgFeatureGrid/AgFeatureGrid.jsx

```diff
@@ -0,0 +1,1021 @@
+      onRowClick(row, feature);
+    }
+
+    this.zoomToFeatures([feature]);
+  }
+
+  /**
+   * Called on row mouseover and hightlights the corresponding feature's
+   * geometry.
+   *
+   * @param {Object} row The highlighted row.
+   */
+  onRowMouseOver = evt => {
+    const {
+      onRowMouseOver
+    } = this.props;
+
+    const row = evt.data;
+    const feature = this.getFeatureFromRowKey(row.key);
+
+    if (isFunction(onRowMouseOver)) {
+      onRowMouseOver(row, feature);
+    }
+
+    this.highlightFeatures([feature]);
+  }
+
+  /**
+   * Called on mouseout and unhightlights any highlighted feature.
+   *
+   * @param {Object} row The unhighlighted row.
+   */
+  onRowMouseOut = evt => {
+    const {
+      onRowMouseOut
+    } = this.props;
+
+    const row = evt.data;
+    const feature = this.getFeatureFromRowKey(row.key);
+
+    if (isFunction(onRowMouseOut)) {
+      onRowMouseOut(row, feature);
+    }
+
+    this.unhighlightFeatures([feature]);
+  }
+
+  /**
+   * Fits the map's view to the extent of the passed features.
+   *
+   * @param {ol.Feature[]} features The features to zoom to.
+   */
+  zoomToFeatures = features => {
+    const {
+      map
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    let featGeometries = [];
+    features.forEach(feature => {
+      featGeometries.push(feature.getGeometry());
+    });
+
+    if (featGeometries.length > 0) {
+      const geomCollection = new OlGeomGeometryCollection(featGeometries);
+      map.getView().fit(geomCollection.getExtent());
+    }
+  }
+
+  /**
+   * Highlights the given features in the map.
+   *
+   * @param {ol.Feature[]} highlightFeatures The features to highlight.
+   */
+  highlightFeatures = highlightFeatures => {
+    const {
+      map,
+      highlightStyle
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    highlightFeatures.forEach(feature => feature.setStyle(highlightStyle));
+  }
+
+  /**
+   * Unhighlights the given features in the map.
+   *
+   * @param {ol.Feature[]} unhighlightFeatures The features to unhighlight.
+   */
+  unhighlightFeatures = unhighlightFeatures => {
+    const {
+      map,
+      selectStyle
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    const selectedRowKeys = this.getSelectedRowKeys();
+
+    unhighlightFeatures.forEach(feature => {
+      const key = this.props.keyFunction(feature);
+      if (selectedRowKeys.includes(key)) {
+        feature.setStyle(selectStyle);
+      } else {
+        feature.setStyle(null);
+      }
+    });
+  }
+
+  /**
+   * Sets the select style to the given features in the map.
+   *
+   * @param {ol.Feature[]} features The features to select.
+   */
+  selectFeatures = features => {
+    const {
+      map,
+      selectStyle
+    } = this.props;
+
+    if (!(map instanceof OlMap)) {
+      return;
+    }
+
+    features.forEach(feature => feature.setStyle(selectStyle));
```

Comment:
Plz make use of a `selectLayer`. See comment above regarding the `highlightLayer`...

In the select case it is even more important...